test/Interop/Cxx/function/emit-called-function-irgen.swift

```diff
@@ -0,0 +1,20 @@
+// Test that Swift emits code for a C++ inline function that is called from
+// another C++ inline function, but not directly from Swift code.
+//
+// We should not, however, emit code for functions that aren't called even
+// indirectly from Swift.
+
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -I %S/Inputs -enable-cxx-interop -emit-ir -o - | %FileCheck %s
```

Comment:
WDYT about running this test in both C and C++ modes?

Reply:
Good point. Done.

I'm using `static inline` in C, as `extern inline` follows a [different model](https://gustedt.wordpress.com/2010/11/29/myth-and-reality-about-inline-in-c99/), where there's always one translation unit that explicitly causes an out-of-line instance to be emitted (so it doesn't have the issue we're looking at here).

`static inline` C functions aren't affected by the bug I'm fixing here as they don't have external linkage, so they don't fall into the case that does the early-out. It's still good to have this test in place though.